server/mbta_api.py

```diff
@@ -89,14 +103,50 @@ async def vehicle_data_for_routes(route_ids, test_mode=False):
             "include": "stop,trip.route_pattern.name",
         },
     )
-    # Iterate vehicles, only send new ones to the browser
+
+    # intialize empty list of vehicles to display
     vehicles_to_display = []
+
+    # intialize dictionary of stats
+    vehicle_stats = {
+        "green_new_active": 0,
+        "orange_new_active": 0,
+        "red_new_active": 0,
+        "green_total_active": 0,
+        "orange_total_active": 0,
+        "red_total_active": 0,
+        "green_percent_active_new": 0,
+        "orange_percent_active_new": 0,
+        "red_percent_active_new": 0,
+    }
+
+    # iterate over all vehicles fetched from V3 API
     for vehicle in vehicles:
         try:
+            # derive Red Line vehicle branch if needed
             custom_route = derive_custom_route_name(vehicle)
-            is_new = fleet.car_array_is_new(custom_route, vehicle["label"].split("-"), test_mode)
-            if not is_new:
+
+            # determine if vehicle is new
+            is_new = fleet.vehicle_array_is_new(custom_route, vehicle["label"].split("-"))
+
+            if "Green" in vehicle["route"]["id"]:
+                vehicle_stats["green_total_active"] += 1
+                if is_new:
+                    vehicle_stats["green_new_active"] += 1
+            if "Orange" in vehicle["route"]["id"]:
+                vehicle_stats["orange_total_active"] += 1
+                if is_new:
+                    vehicle_stats["orange_new_active"] += 1
+            if "Red" in vehicle["route"]["id"]:
+                vehicle_stats["red_total_active"] += 1
+                if is_new:
+                    vehicle_stats["red_new_active"] += 1
+
+            # if not running test mode and vehicle is not new, skip this vehicle
+            if not test_mode and not is_new:
```

Comment:
eep- this function is doing too much. Can we move stat calculations to a sub call?

maybe `vehicle["label"].split("-")` can be moved into `vehicle_array_is_new` (and maybe rename the fn then)?
I would probably do these calculations with something like `green_vehs = (list(filter(is_green, vehicles))); green_count = len(green_vehs)` etc.
Seems more pythonic, but it's probably just a personal preference. What you have now is perfectly readable (if a little verbose)

Reply:
Yeah this was a late-night addition - happy to make changes to it. @idreyn any thoughts on this as well?

Reply:
Maybe it could be structured like:
```py
vehicle_stats = {
    "Orange": {
        "totalActive": 20,
        "totalNew": 3,
    },
    ...
}
```
Eventually on the client we'll need to break these apart by line.